### PR TITLE
Add icons for new sensor device classes

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Add icons for new sensor device classes

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/const.ts [hunk 1]
```diff
@@ -121,6 +122,7 @@ export const FIXED_DEVICE_CLASS_ICONS = {
   carbon_monoxide: mdiMoleculeCo,
   current: mdiCurrentAc,
   date: mdiCalendar,
+  distance: mdiSpeedometer,
```
Comment (epenet) [inline src/common/const.ts:125]: Would ruler not be better for distance?
```suggestion
  distance: mdiRuler,
```

File: src/common/const.ts [hunk 2]
```diff
@@ -140,11 +142,13 @@ export const FIXED_DEVICE_CLASS_ICONS = {
   pressure: mdiGauge,
   reactive_power: mdiFlash,
   signal_strength: mdiWifi,
+  speed: mdiSpeedometer,
   sulphur_dioxide: mdiMolecule,
   temperature: mdiThermometer,
   timestamp: mdiClock,
   volatile_organic_compounds: mdiMolecule,
   voltage: mdiSineWave,
+  // volume: TBD, => no well matching icon found
```
Comment (epenet) [inline src/common/const.ts:151]: Not ideal, but maybe `mdiCupWater` as default?

File: src/common/const.ts (hunk 1 above)
Comment (frenck) [reply src/common/const.ts:125]: This is an error :) 
We also discussed using:

CleanShot 2022-09-28 at 09 40 31

Comment (emontnemery) [reply src/common/const.ts]: I updated the icon to what was intended, but I agree @epenet's suggestion would work too:
image
Or maybe:
image

File: src/common/const.ts (hunk 2 above)
Comment (emontnemery) [reply src/common/const.ts:152]: A very quick browse of the use `VOLUME_*` constants shows that, where the context is clear, it's mostly either of:
- gas
- water
- vehicle fuel
For gas, there's already a gas device class of course, so it's pretty much either an amount of water or vehicle fuel

The cup is not a great fit for vehicle fuel.

How about just doing this for the default:
image

Comment (emontnemery) [reply src/common/const.ts:152]: There's also:
image

Comment (epenet) [reply src/common/const.ts:152]: I like that `mdiCarCoolantLevel` shows inside a container.
image

Comment (emontnemery) [reply src/common/const.ts:152]: It's not a good fit for water or fuel consumption though.

Comment (frenck) [reply src/common/const.ts:153]: Let's hold off an icon and let's see over time what is being used by integrations.

Comment (emontnemery) [reply src/common/const.ts:153]: Yes, it fits an amount of liquid in a container, but it doesn't fit the amount of water you've used in your home, or the amount of fuel your car has used.

**Review by frenck — APPROVED** [review]
I think this is OK to start with. We can always improve/change based on feedback provided by others in the future. Beta is close, and we should get some default in.

**Comment by epenet** [comment]
Hi
I was surprised by the comment from @balloob on this PR: https://github.com/home-assistant/core/pull/79323, asking the user to drop the icon and effectively be left without a default icon.

Opposite, I left the existing icons in my PRs: https://github.com/home-assistant/core/pull/79261, https://github.com/home-assistant/core/pull/79263, https://github.com/home-assistant/core/pull/79262, https://github.com/home-assistant/core/pull/79253, https://github.com/home-assistant/core/pull/79277, https://github.com/home-assistant/core/pull/79285, https://github.com/home-assistant/core/pull/77488

Same for @bdraco in https://github.com/home-assistant/core/pull/79162

Do we need to go over these PRs a second time?

**Comment by bdraco** [comment]
Icon to use in the frontend. Icons start with mdi: plus an identifier. You probably don't need this since Home Assistant already provides default icons for all entities according to its device_class. This should be used only in the case where there either is no matching device_class or where the icon used for the device_class would be confusing or misleading.

https://developers.home-assistant.io/docs/core/entity/

If the icon provided by the device class is not confusing or misleading it should be removed.  If there is no icon provided by the device class it should likely still be set.

**Comment by epenet** [comment]
I reviewed my changes, and opened this PR: https://github.com/home-assistant/core/pull/79348

**Pull request merged**